From d7f4f51d3e508e44c706e2a3f4d2fe0f695cbc0e Mon Sep 17 00:00:00 2001
From: Robbie Cao <robbie.cao@arm.com>
Date: Thu, 26 Oct 2023 12:10:47 +0800
Subject: [PATCH] Plat: kronos: Enable capsule update service

Add capsule update service component on kronos platform. Also add
function to initialize the capsule update provider.

Signed-off-by: Robbie Cao <robbie.cao@arm.com>
Signed-off-by: Qi Feng <qi.feng@arm.com>
Upstream-Status: Inappropriate [Design is to revisted]
---
 deployments/se-proxy/infra/kronos/infra.cmake  |  1 +
 .../infra/kronos/service_proxy_factory.c       | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/deployments/se-proxy/infra/kronos/infra.cmake b/deployments/se-proxy/infra/kronos/infra.cmake
index e6c363d..88906d8 100644
--- a/deployments/se-proxy/infra/kronos/infra.cmake
+++ b/deployments/se-proxy/infra/kronos/infra.cmake
@@ -24,6 +24,7 @@ add_components(TARGET "se-proxy"
 	"components/service/crypto/backend/psa_ipc"
 	"components/service/crypto/client/psa"
 	"components/service/secure_storage/backend/mock_store"
+	"components/service/capsule_update/provider"
 )
 
 target_sources(se-proxy PRIVATE
diff --git a/deployments/se-proxy/infra/kronos/service_proxy_factory.c b/deployments/se-proxy/infra/kronos/service_proxy_factory.c
index 154558e..f4a30c3 100644
--- a/deployments/se-proxy/infra/kronos/service_proxy_factory.c
+++ b/deployments/se-proxy/infra/kronos/service_proxy_factory.c
@@ -14,6 +14,7 @@
 #include <service/crypto/factory/crypto_provider_factory.h>
 #include <service/secure_storage/frontend/secure_storage_provider/secure_storage_provider.h>
 #include <trace.h>
+#include <service/capsule_update/provider/capsule_update_provider.h>
 
 struct rss_comms_caller rss_comms;
 
@@ -93,3 +94,20 @@ struct rpc_interface *its_proxy_create(void)
 
 	return secure_storage_provider_init(&its_provider, backend);
 }
+
+struct rpc_interface *capsule_update_proxy_create(void)
+{
+	static struct capsule_update_provider capsule_update_provider;
+	static struct rpc_caller *capsule_update_caller;
+
+	capsule_update_caller = rss_comms_caller_init(&rss_comms);
+
+	if (!capsule_update_caller) {
+		EMSG("capsule update caller is null\n");
+		return NULL;
+	}
+
+	capsule_update_provider.client.caller = capsule_update_caller;
+
+	return capsule_update_provider_init(&capsule_update_provider);
+}
-- 
2.25.1

